src/extension/geofence-server/src/test/java/org/geoserver/geofence/web/GeofenceRulePageTest.java

```diff
@@ -0,0 +1,66 @@
+/* (c) 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.geofence.web;
+
+import static org.junit.Assert.*;
+
+import java.util.logging.Level;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+import org.apache.wicket.markup.html.form.Form;
+import org.apache.wicket.util.tester.FormTester;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.web.data.layergroup.LayerGroupBaseTest;
+import org.junit.Test;
+import org.opengis.referencing.FactoryException;
+
+public class GeofenceRulePageTest extends LayerGroupBaseTest {
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        login();
+        super.onSetUp(testData);
+    }
+
+    @Test
+    public void testAddLayerGroupRule() throws FactoryException {
```

Comment:
Weak testing, should verify various layer group modes and layers contained in multiple groups. Probably better done in "AccessManagerTest" though, as you can verify how the final access rules contain.

Reply:
I've added a GeofenceAccessManagerIntegrationTest class to tests the accessManager behaviour with direct access and multiple containers. Also a GetMapIntegration class was already present in the first commit of this pull request, showing tests for SINGLE and OPAQUE containers, with and without allowed areas